snapshotter/demux/cache/snapshotter_cache.go

```diff
@@ -25,18 +25,20 @@ import (
 // SnapshotterCache implements a read, write protected cache mechanism
 // for keyed snapshotters.
 type SnapshotterCache struct {
-	mutex        *sync.Mutex
+	mutex        *sync.RWMutex
 	snapshotters map[string]snapshots.Snapshotter
 }
 
 // NewSnapshotterCache creates a new instance with an empty cache.
 func NewSnapshotterCache() *SnapshotterCache {
-	return &SnapshotterCache{&sync.Mutex{}, make(map[string]snapshots.Snapshotter)}
+	return &SnapshotterCache{&sync.RWMutex{}, make(map[string]snapshots.Snapshotter)}
 }
 
 // Get fetches and caches the snapshotter for a given key.
 func (cache *SnapshotterCache) Get(ctx context.Context, key string, fetch SnapshotterProvider) (snapshots.Snapshotter, error) {
+	cache.mutex.RLock()
```

Comment:
This was specifically setup this way to not need a read lock here. Did something change with this PR that makes it necessary?

Reply:
Hmm I suppose not. My thinking was the read lock in `Get` would be necessary for read consistency... but given that on a cache miss, a write lock is acquired and used for both reading and potentially writing to the cache, this read lock is not necessary.